runtime/lua/vim/lsp/structures/init.lua

```diff
@@ -0,0 +1,36 @@
+--[[
+
+structures.lua
+
+structures is the location to put the handling of any of the structures as defined by
+the language server protocol. Any function defined on a structure must take as the first
+argument the structure itself and then any optional parameters required for determining
+the behavior.
+
+For example, all `Location.*` must be a function of the form:
+
+  function(location, ...)
+
+Additionally, within each `structure.*`, they can import only directly from other structures using:
```

Comment:
This note should be for developers only (or noted somewhere in dev docs) but it useless for end users and I dont' want to show the wrong imports

Reply:
Yes, within the structures file, it will cause circular dependencies. This prevents that. But from outside of the module, you should import from the init in case there is any setup / teardown / etc. that needs to happen at the module level.

Perhaps I will move them all to `structures._Location` for example. What do you think of that?